lib/promisify.ts

```diff
@@ -1,7 +1,8 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 const isFn = (obj: unknown) => typeof obj === 'function';
 
-const isAsyncFn = (fn: (...args: any[]) => any) => fn.prototype.constructor.name.endsWith('Async');
+const isAsyncFn = (fn: (...args: any[]) => any) =>
+  fn.prototype?.constructor?.name?.endsWith('Async');
```

Comment:
I'm not a big fan of the ?. syntax, I'm still waiting for a wider adoption to use it.

Reply:
maybe you could also use less than 3? to make your fix work.

Reply:
`fn.prototype.constructor?.name.endsWith('Async');` should works fine

Reply:
```
(fn: (...args: any[]) => any) => {
  const { constructor } = fn.prototype;
  if (!constructor) return false;
  return constructor.name.endsWith('Async');
}
```

Reply:
or a simplest way:
```ts
(fn: (...args: any[]) => any) => {
  try {
    return  fn.prototype?.constructor?.name?.endsWith('Async');
  } catch (e) {
    return false;
  }
}
```
- easy to read
- do not user `?.`
- TS and JS code is the same.

Reply:
I'll use the `try`/`catch` version.